app/src/ui/app.tsx

```diff
@@ -583,6 +585,23 @@ export class App extends React.Component<IAppProps, IAppState> {
     this.props.dispatcher.setAppMenuState(menu => menu.withReset())
   }
 
+  private onTitlebarDoubleClickDarwin() {
+    const actionOnDoubleClick: AppleActionOnDoubleClickPref = remote.systemPreferences.getUserDefault('AppleActionOnDoubleClick', 'string')
+    const mainWindow = remote.getCurrentWindow()
+
+    switch (actionOnDoubleClick) {
+      case 'Maximize':
+        mainWindow.isMaximized() ? mainWindow.unmaximize() : mainWindow.maximize()
+        break
+      case 'Minimize':
+        mainWindow.minimize()
+        break
+      case 'None':
+      default:
```

Comment:
Seeing how we're not doing anything after this I'd be in favor of skipping it completely but I'll let @shiftkey make the final call as I believe he was the one who suggested having a default.